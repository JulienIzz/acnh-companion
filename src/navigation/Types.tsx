export type RootStackParamList = {
  Home: undefined;
  Encyclopedie: undefined;
  Search: undefined;
  BugReporting: undefined;
};

export type Routes = keyof RootStackParamList;

declare global {
  namespace ReactNavigation {
    interface RootParamList extends RootStackParamList {}
  }
}
