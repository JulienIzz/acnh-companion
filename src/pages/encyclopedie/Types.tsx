export interface Fish {
  id: number;
  name: {
    'name-USen': string;
    'name-EUfr': string;
  };
  availability: {
    'month-northern': string;
    'month-southern': string;
    time: string;
    isAllDay: boolean;
    isAllYear: boolean;
    location: string;
    rarity: string;
    'month-array-northern': Array<number>;
    'month-array-southern': Array<number>;
  };
  shadow: string;
  price: number;
  'price-cj': number;
  icon_uri: string;
  image_uri: string;
}
