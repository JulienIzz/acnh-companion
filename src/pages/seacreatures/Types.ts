export type SeaCreature = {
  id: number;
  name: {
    'name-USen': string;
    'name-EUfr': string;
  };
  availability: {
    'month-northern': string;
    'month-southern': string;
    time: string;
    isAllDay: boolean;
    isAllYear: false;
    'month-array-northern': Array<number>;
    'month-array-southern': Array<number>;
    'time-array': Array<number>;
  };
  speed: string;
  shadow: string;
  price: number;
  image_uri: 'https://acnhapi.com/v1/images/sea/1';
  icon_uri: 'https://acnhapi.com/v1/icons/sea/1';
};
