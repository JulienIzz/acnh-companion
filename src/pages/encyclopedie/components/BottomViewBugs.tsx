import React from 'react';
import {SimpleTypography} from '../../../theme/typography/Typography';
import {Bug} from '../Types';

interface BottomViewBugsProps {
  animal: Bug;
}

export const BottomViewBugs = ({animal}: BottomViewBugsProps) => {
  return (
    <>
      <SimpleTypography
        text={'DisponibilitÃ© : ' + animal.availability['month-northern']}
      />
      <SimpleTypography text={'Horaire : ' + animal.availability.time} />
      <SimpleTypography
        text={'Localisation : ' + animal.availability.location}
      />
    </>
  );
};
