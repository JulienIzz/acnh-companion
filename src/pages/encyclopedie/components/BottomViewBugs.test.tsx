import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {BottomViewBugs} from './BottomViewBugs';
import {Bug} from '../Types';

const mockFishArray: Bug[] = [
  {
    id: 1,
    name: {
      'name-USen': 'PoissonPilote1',
      'name-EUfr': 'PoissonPilote1',
    },
    availability: {
      'month-northern': 'Avril à Septembre',
      'month-southern': 'Avril à Septembre',
      'month-array-northern': [1, 2],
      'month-array-southern': [1, 2],
      time: '9am - 4pm',
      isAllDay: false,
      isAllYear: false,
      location: 'River',
      rarity: 'Common',
    },
    'price-cj': 1200,
    icon_uri: '',
    image_uri: '',
    price: 900,
  },
];

test('animal bottom view card with mock bug information', async () => {
  render(
    <BottomViewBugs
      months={mockFishArray[0].availability['month-northern']}
      hours={mockFishArray[0].availability.time}
      location={mockFishArray[0].availability.location}
    />,
  );

  const monthsOutput = screen.getByText('Disponibilité : Avril à Septembre');
  const hoursOutput = screen.getByText('Horaire  : 9am - 4pm');
  const locationOutput = screen.getByText('Localisation : River');

  expect(monthsOutput).toBeTruthy();
  expect(hoursOutput).toBeTruthy();
  expect(locationOutput).toBeTruthy();

  expect(screen.toJSON()).toMatchSnapshot();
});
