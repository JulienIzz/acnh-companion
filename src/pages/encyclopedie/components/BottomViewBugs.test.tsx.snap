// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`animal bottom view card with mock bug information 1`] = `
[
  <Text
    style={
      {
        "color": "black",
        "fontSize": 14,
      }
    }
  >
    Disponibilité : Avril à Septembre
  </Text>,
  <Text
    style={
      {
        "color": "black",
        "fontSize": 14,
      }
    }
  >
    Horaire : 9am - 4pm
  </Text>,
  <Text
    style={
      {
        "color": "black",
        "fontSize": 14,
      }
    }
  >
    Localisation : River
  </Text>,
]
`;
